openvm/src/bus_interaction_handler/mod.rs

```diff
@@ -91,6 +112,131 @@ impl<T: FieldElement> IsBusStateful<T> for OpenVmBusInteractionHandler<T> {
     }
 }
 
+impl<T: FieldElement> RangeConstraintHandler<T> for OpenVmBusInteractionHandler<T> {
+    fn pure_range_constraints<V: Ord + Clone + Eq>(
+        &self,
+        bus_interaction: &BusInteraction<GroupedExpression<T, V>>,
+    ) -> Option<RangeConstraintMap<T, V>> {
+        let bus_id = bus_interaction
+            .bus_id
+            .try_to_number()
+            .unwrap()
+            .to_integer()
+            .try_into_u64()
+            .unwrap();
+        match self.bus_map.bus_type(bus_id) {
+            BusType::ExecutionBridge | BusType::Memory | BusType::PcLookup => None,
```

Comment:
Shouldn't memory bus interactions also give byte constraints to the data?

Reply:
They do return byte constraints in the implementation of `BusInteractionHandler`, but they are not pure range constraints, so returning `None` here is correct.